### PR TITLE
[DataTable] Move page size options to parent

### PR DESCRIPTION
**Pull request by SferaDev**
Introducing the prop change we spoke in the last weekly meeting @tokland, will be used for ``metadata-sync`` new widget.

### CODE REVIEW

**Comment by adrianq** [comment]
@tokland have you tested this PR? Should we merge it for the next release?

**Comment by tokland** [comment]
> @tokland have you tested this PR? Should we merge it for the next release?

Not yet. We don't need it for project-monitoring. I can test it tomorrow, though.

**Comment by adrianq** [comment]
Will merge the other PRs and wait for your feedback on this one. Thanks!

File: src/data-table/DataTable.tsx [hunk 1]
```diff
@@ -117,6 +118,7 @@ export function DataTable<T extends ReferenceObject = TableObject>(props: DataTa
         sorting: controlledSorting,
         selection: controlledSelection,
         pagination: controlledPagination,
+        pageSizeOptions = [10, 25, 50, 100],
```
Comment (tokland) [inline src/data-table/DataTable.tsx]: Typically, we'd like to set also the default value. Maybe we can expand that prop: `paginationOptions={{pageSizeOptions: [10, 20], pageSizeInitialValue: 20}}`, or something like that?

File: src/data-table/DataTable.tsx [hunk 2]
```diff
@@ -138,13 +140,15 @@ export function DataTable<T extends ReferenceObject = TableObject>(props: DataTa
         updateVisibleColumns(visibleColumns => _.uniq([...visibleColumns, ...newVisibleColumns]));
     }, [columns]);
 
+    const pageSize = pageSizeOptions.includes(25) ? 25 : pageSizeOptions[0];
```
Comment (tokland) [inline src/data-table/DataTable.tsx]: ... and then we can remove this magic 25.

**Comment by SferaDev** [comment]
I will update with your proposal @tokland, for now I'm updating the beta so it doesn't break the current PR

**Comment by adrianq** [comment]
@tokland It seems to be working fine in metadata sync. Let me know if it works for you and I will merge it...

**Review by tokland — APPROVED** [review]
It works on project-monitoring-app. I just changed the type to Partial, as it's used this way.

**Pull request merged**